R/bfast01classify.R

```diff
@@ -64,51 +103,136 @@ bfast01classify <- function(object, alpha=0.05, pct_stable=NULL) {
+   }
+   
+   ## CLASSIFICATION 
+   ## Standard Typo
+   if(typology=="standard"){
+      ## monotonic if no break
+      if(object$breaks == 0) {     
+         slope <- object$model[[1]]$coefficients[2] # slope
+         if(slope > 0) out[1] <- 1
+         if(slope < 0) out[1] <- 2
+      }else{
+         ## classes with break
+         # with break, but still monotonic
+         if(s1 > 0 && s2 > 0 && m > 0) out[1] <- 3
+         if(s1 < 0 && s2 < 0 && m < 0) out[1] <- 4
+         # interrupted gradual change (setback or boost)
+         if(s1 > 0 && s2 > 0 && m < 0) out[1] <- 5
+         if(s1 < 0 && s2 < 0 && m > 0) out[1] <- 6
+         # trend reversal (greening to browning v.v.)
+         if(s1 > 0 && s2 < 0) out[1] <- 7
+         if(s1 < 0 && s2 > 0) out[1] <- 8
+      }
+   }   
+      
+   # Drylands Typo
+   if(typology=="drylands"){
+      ## monotonic if no break
+      if(object$breaks == 0) {
+         if(p.seg[1] > alpha){
+         out[1] <- 0 #fluctuating/no change
+      }else{
+         slope <- object$model[[1]]$coefficients[2] # slope
+         if(slope > 0) out[1] <- 1 # monotonic increase
+         if(slope < 0) out[1] <- 2 # monotonic decrease  
+      }
+      } else {
+         ## classes with break
+         if(p.seg[1] > alpha && p.seg[2] > alpha){
+            out[1] <- 0 #fluctuating/no change
+         }else{
+            # with break, but still same direction
+            if(s1 > 0 && s2 > 0) out[1] <- 3 # non-monotonic increase
+            if(s1 < 0 && s2 < 0) out[1] <- 4 # non-monotonic decrease
+            # trend reversal (greening to browning v.v.)
+            if(s1 > 0 && s2 < 0) out[1] <- 5 # reversal increase to decrease
+            if(s1 < 0 && s2 > 0) out[1] <- 6 # reversal decrease to increase
+         }
+      }
+  
+      ## Sub-classification
+      if(object$breaks != 0 && out[1] != 0) { #if there's a break and at least one sign. trend
+         if( (s1 > 0 && s2 > 0) || (s1 < 0 && s2 < 0)){
+         # non mono trends (sub-types 1 and 2, "slowing down" and "accelerating")
+         if( (s1 > 0 && s2 > 0) || (s1 < 0 && s2 < 0) ){
+            if(abs(s1) > abs(s2)) out[2] <- 1 # slowing down
+            if(abs(s1) < abs(s2)) out[2] <- 2 # accelerating
+         }
+      }else if( (s1 < 0 && s2 > 0) || (s1 > 0 && s2 < 0)){
+         # reversal trend (sub-types 3 and 4, "transition" and "complete")
+         if( (p.seg[1] <= alpha && p.seg[2] > alpha) || (p.seg[1] > alpha && p.seg[2] <= alpha) ) out[2] <- 3 # transition
+         if(p.seg[1] <= alpha && p.seg[2] <= alpha) out[2] <- 4 # complete
+         }
+      }
    }
+    
    ## Segment significance flag
    # code: 0 = both segments significant (or no break and significant), 
    # 1 = only first segment significant, 
    # 2 = only 2nd segment significant, 
    # 3 = both segments insignificant (or no break and not significant)
    
+   if(typology=="standard"){ isf = 0 }
+   if(typology=="drylands"){ isf = 1 }
```

Comment:
There's a lot of index wrangling throughout. Wouldn't it be simpler if you added the subtype flag to the end of the list? To reorder the output, you could just use a single line before `return`ing the result.

Reply:
Indeed, reordering the output just before the end would make the code simpler. I'll make the necessary changes. At first I was putting the subtype flag in the end of the list, but as it's a really relevant information of the new typology, I thought it would be better to present it right after the type flag.